testing-modules/mockito-2/src/test/java/com/baeldung/mockito/additionalanswers/BookServiceUnitTest.java

```diff
@@ -65,4 +72,27 @@ public void givenCheckifEqualsMethodMocked_whenCheckifEqualsInvoked_ThenReturnAr
 
         assertEquals(bookOnIndex, book2);
     }
+
+    @Test
+    public void givenMockedMethod_whenMethodInvoked_thenReturnBook() {
+        Long id = 1L;
+        when(bookRepository.getByBookId(anyLong())).thenAnswer(AdditionalAnswers.answer((Long bookId) -> new Book(bookId, "The Stranger", "Albert Camus", 456)));
+
+        assertNotNull(bookService.getByBookId(id));
+        assertEquals("The Stranger", bookService.getByBookId(id).getTitle());
+    }
+    
+    @Test
+    public void givenMockedMethod_whenMethodInvoked_thenReturnVoid() {
+        Long id = 2L;
+        when(bookRepository.getByBookId(anyLong())).thenAnswer(AdditionalAnswers.answerVoid(new VoidAnswer1<Long>() {
+            public void answer(Long id) {
+                System.out.println(id);
+            }
+        }));
```

Comment:
Hi - I created this feature in Mockito, and I also am an editor at Baeldung.

It's not meant to be used with anonymous inner classes. It was there to make it easy to plug a lambda into mockito.

```java
when(...)
   .thenAnswer(answerVoid((Long id) -> { 
      System.out.println(id);
   });
```

Note also the use of static imports to make the code readable.

It's worth noting that this may get a bit harder for the compiler, so it's sometimes a good idea to create a static method inside the test class with the right signature and then refer to it with a method reference.